### PR TITLE
mongodb_user.py catch error return in users and display error.

### PR DESCRIPTION
**Pull request by diepes**
### What does this PR do?
Catch a error returned in users, exiting with a error rather than trying to loop through the list.

### Previous Behavior
Salt would exit with a error
"AttributeError: 'bool' object has no attribute 'get'"

### New Behavior
Salt exits with a error
Result: False
Comment: Mongo Err: not authorized on admin to execute command { usersInfo: "root" }

### Tests written?

No

### Commits signed with GPG?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by DmitryKuzmenko — APPROVED** [review]

File: salt/states/mongodb_user.py [hunk 1]
```diff
@@ -113,6 +113,13 @@ def present(name,
     if len(users) > 0:
         # check each user occurrence
         users = __salt__['mongodb.user_find'](name, user, password, host, port, database, authdb)
+        # catch errors e.g.
+        # users= (False, 'Failed to connect to MongoDB database localhost:27017')
+        # users= (False, 'not authorized on admin to execute command { usersInfo: "root" }')
+        if not users[0]:
+            ret['result'] = False
+            ret['comment']="Mongo Err: "+str(users[1])
```
Comment (DmitryKuzmenko) [inline salt/states/mongodb_user.py]: Please surround the `=` with spaces to fix lint errors.

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]
Thank you for PR. Looks good, it's only needed to fix lint errors.

**Comment by cachedout** [comment]
@diepes This needs a rebase, please. There are merge conflicts.

**Comment by diepes** [comment]
Re-based and fixed the lint

File: salt/states/mongodb_user.py (hunk 1 above)
Comment (diepes) [reply salt/states/mongodb_user.py]: Done.

**Review by diepes — COMMENTED** [review]
Change done, spaces added.

File: salt/states/mongodb_user.py (hunk 1 above)
Comment (diepes) [reply salt/states/mongodb_user.py]: Fixed.

**Review by diepes — COMMENTED** [review]
Changes made to fix lint.

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
Thank you for this fix @diepes and welcome! Apologies for taking so long to get to this PR. I have kicked off another test run. Once those are finished, we can get this in.

**Pull request merged**